### PR TITLE
[SCB-2422]change bean injection to constructor injection

### PR DESCRIPTION
**Pull request by YOC19970520**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Optimized the bean injection with constructor-based injection and setter injection. This PR might has some problems. Please comment on it and I will modify it again if some code change in a wrong injection way.

### CODE REVIEW

File: governance/src/main/java/org/apache/servicecomb/governance/properties/GovernanceProperties.java [hunk 1]
```diff
@@ -53,12 +53,17 @@
...
 
-  @Autowired
   private MicroserviceMeta microserviceMeta;
 
+  @Autowired
+  public GovernanceProperties(String configKey, Environment environment, MicroserviceMeta microserviceMeta) {
+    this.configKey = configKey;
+    this.environment = environment;
+    this.microserviceMeta = microserviceMeta;
+  }
```
Comment (liubao68) [inline governance/src/main/java/org/apache/servicecomb/governance/properties/GovernanceProperties.java:65]:
```
  protected GovernanceProperties(String key) {
    configKey = key;
    representer.getPropertyUtils().setSkipMissingProperties(true);
    GovernanceEventManager.register(this);
    entityClass = getEntityClass();
  }
```

是否删除？ 另外 @Autowired 是否加到 property 上面？ 因为 configKey 是无法autowired。

Comment (YOC19970520) [reply governance/src/main/java/org/apache/servicecomb/governance/properties/GovernanceProperties.java:65]:
```  
    protected GovernanceProperties(String key) {
    configKey = key;
    representer.getPropertyUtils().setSkipMissingProperties(true);
    GovernanceEventManager.register(this);
    entityClass = getEntityClass();
  }
```
不删除,该构造方法里初始化了configKey; 将@Autowired改成setter方法注入,只注入environment和microserviceMeta

**Comment by Shoothzj** [comment]
we should use `[` instead of chinese `【`

**Comment by YOC19970520** [comment]
Yes sir. I'll modified it tomorrow. Thanks for reminding.



---Original---
From: "ZhangJian ***@***.***&gt;
Date: Wed, Mar 23, 2022 19:12 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [apache/servicecomb-java-chassis] 【SCB-2422】change bean injection to constructor injection (PR #2736)




 
we should use [ instead of chinese 【

File: governance/src/main/java/org/apache/servicecomb/router/distribute/AbstractRouterDistributor.java [hunk 2]
```diff
@@ -46,9 +46,16 @@
...
-  @Autowired
   private RouterRuleCache routerRuleCache;
 
+  @Autowired
+  public AbstractRouterDistributor(RouterRuleCache routerRuleCache) {
+    this.routerRuleCache = routerRuleCache;
+  }
+
+  protected AbstractRouterDistributor() {
+  }
```
Comment (liubao68) [inline governance/src/main/java/org/apache/servicecomb/router/distribute/AbstractRouterDistributor.java:57]: change to setter injection

File: pom.xml [hunk 3]
```diff
@@ -519,6 +519,17 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-maven-plugin</artifactId>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <configuration>
+              <skip>true</skip>
+            </configuration>
+          </plugin>
```
Comment (liubao68) [inline pom.xml:532]: Why add there two plugins?

**Review by liubao68 — APPROVED** [review]

**Pull request merged**